pkg/approver/aws/approver.go

```diff
@@ -0,0 +1,139 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/juju/ratelimit"
+
+	certificates "k8s.io/api/certificates/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	certificatesv1b1informers "k8s.io/client-go/informers/certificates/v1beta1"
+	"k8s.io/client-go/kubernetes"
+	certificatesv1b1listers "k8s.io/client-go/listers/certificates/v1beta1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+)
+
+// Config defines settings for Approver.
+type Config struct {
+	RegionName  string
+	AllowedASGs []string
+}
+
+// Approver approvers CSRs.
+type Approver struct {
+	kubeClient  kubernetes.Interface
+	aws         *awsCloud
+	allowedASGs sets.String
+
+	csrLister certificatesv1b1listers.CertificateSigningRequestLister
+	csrSynced cache.InformerSynced
+	queue     workqueue.RateLimitingInterface
+}
+
+// New return a new Approver.
+func New(client kubernetes.Interface, csrInformer certificatesv1b1informers.CertificateSigningRequestInformer, config Config) (*Approver, error) {
```

Comment:
The reason the other approver takes an informer is because it's part of the shared informer. I think you can safely construct it form the `client kubernetes.Interface` here.

Reply:
done